cmd/image-builder/main.go

```diff
@@ -64,56 +46,55 @@ func main() {
 		panic(err)
 	}
 
-	if conf.GlitchTipDSN != "" {
-		err = sentry.Init(sentry.ClientOptions{
-			Dsn: conf.GlitchTipDSN,
-		})
-		if err != nil {
-			panic(err)
-		}
-	}
-
-	err = logger.ConfigLogger(logrus.StandardLogger(), conf.LogLevel)
-	if err != nil {
-		panic(err)
-	}
-	logrus.AddHook(&ctxHook{})
-
-	gitRev, err := gitRev()
+	hostname, err := os.Hostname()
 	if err != nil {
-		logrus.Warn(err.Error())
-		gitRev = "unknown"
+		hostname = "image-builder-unknown"
 	}
 
-	logrus.Infof("Starting image-builder from Git Hash: %s", gitRev)
-	logrus.Infof("Changelog: https://github.com/osbuild/image-builder/commits/%s", gitRev)
-
-	if conf.GlitchTipDSN == "" {
-		logrus.Warn("Sentry/Glitchtip was not initialized")
-	} else {
-		sentryhook := sentrylogrus.NewFromClient([]logrus.Level{logrus.PanicLevel,
-			logrus.FatalLevel, logrus.ErrorLevel},
-			sentry.CurrentHub().Client())
-		logrus.AddHook(sentryhook)
-	}
-
-	if conf.CwAccessKeyID != "" {
-		err = logger.AddCloudWatchHook(logrus.StandardLogger(), conf.CwAccessKeyID, conf.CwSecretAccessKey, conf.CwRegion, conf.LogGroup)
-		if err != nil {
-			panic(err)
-		}
-	}
-
-	if conf.DeploymentChannel != "" {
-		logrus.AddHook(&slogger.EnvironmentHook{Channel: conf.DeploymentChannel})
+	loggingConfig := sinit.LoggingConfig{
+		StdoutConfig: sinit.StdoutConfig{
+			Enabled: true,
+			Level:   "warning",
+			Format:  "text",
+		},
+		SplunkConfig: sinit.SplunkConfig{
+			Enabled:  conf.SplunkHost != "" && conf.SplunkPort != "" && conf.SplunkToken != "",
+			Level:    conf.LogLevel,
+			URL:      fmt.Sprintf("https://%s:%s/services/collector/event", conf.SplunkHost, conf.SplunkPort),
+			Token:    conf.SplunkToken,
+			Source:   "image-builder",
+			Hostname: hostname,
+		},
+		CloudWatchConfig: sinit.CloudWatchConfig{
+			Enabled:      conf.CwAccessKeyID != "" && conf.CwSecretAccessKey != "" && conf.CwRegion != "",
+			Level:        conf.LogLevel,
+			AWSRegion:    conf.CwRegion,
+			AWSSecret:    conf.CwSecretAccessKey,
+			AWSKey:       conf.CwAccessKeyID,
+			AWSLogGroup:  conf.LogGroup,
+			AWSLogStream: hostname,
+		},
+		SentryConfig: sinit.SentryConfig{
+			Enabled: conf.GlitchTipDSN != "",
+			DSN:     conf.GlitchTipDSN,
+		},
+		TracingConfig: sinit.TracingConfig{
+			Enabled: true,
+			CustomAttrs: []slog.Attr{
+				slog.String("channel", "conf.DeploymentChannel"),
+			},
+		},
 	}
```

Comment:
This is the universal configuration block that is used here and in maintainance/migrate commands as well as it will be used in composer/worker. This makes sure logging is always configured in a same way.